jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/status/CodeWhispererStatusBarManager.kt

```diff
@@ -0,0 +1,62 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.codewhisperer.status
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.components.service
+import com.intellij.openapi.extensions.ExtensionPointName
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.wm.StatusBarWidgetFactory
+import com.intellij.openapi.wm.impl.status.widget.StatusBarWidgetSettings
+import com.intellij.openapi.wm.impl.status.widget.StatusBarWidgetsManager
+import software.aws.toolkits.jetbrains.core.credentials.ToolkitConnection
+import software.aws.toolkits.jetbrains.core.credentials.ToolkitConnectionManagerListener
+import software.aws.toolkits.jetbrains.core.credentials.pinning.CodeWhispererConnection
+import software.aws.toolkits.jetbrains.core.credentials.pinning.ConnectionPinningManagerListener
+import software.aws.toolkits.jetbrains.core.credentials.pinning.FeatureWithPinnedConnection
+import software.aws.toolkits.jetbrains.services.codewhisperer.explorer.isCodeWhispererEnabled
+
+/**
+ * Manager visibility of CodeWhisperer status bar widget, only display it when CodeWhisperer is connected
+ */
+@Service(Service.Level.PROJECT)
+class CodeWhispererStatusBarManager(private val project: Project) {
```

Comment:
I'm not too familiar with this codebase creating a separate top-level class for this functionality seems heavy. 

No need to make any changes, this is just an observation. Personally I would've implemented this logic in `CodeWhispererStatusBarWidgetFactory::createWidget` though maybe it would be harder to test?

Reply:
yeaa agree, kinda feel not worth creating another top-level class for this simple functionality, Jaden let me create another SIM tracking this optimization since we wanna roll out this fix tomorrow lol